### PR TITLE
Python metadata fetch

### PR DESCRIPTION
**Pull request by luke777gu**
WIP: will update CI/CD test and fetch metadata for python

### CODE REVIEW

File: .github/workflows/test_pipeline.yml [hunk 1]
```diff
@@ -41,6 +41,7 @@ jobs:
           python -m pip install requests
           python -m pip install datetime
           python -m pip install pypistats
```
Comment (vpopil) [inline .github/workflows/test_pipeline.yml:43]: those need to go into `requirements.txt`  and installed with `... -r requirements.txt`

File: .github/workflows/test_pipeline.yml [hunk 2]
```diff
@@ -51,3 +52,4 @@ jobs:
           DB_NAME: postgres
           DB_HOST: localhost
           GH_TOKEN: ${{ secrets.GH_TOKEN }}
```
Comment (vpopil) [inline .github/workflows/test_pipeline.yml:54]: should this one be deleted?

File: pipeline/scraper/github_query.py [hunk 3]
```diff
@@ -5,14 +5,15 @@
...
 import requests
-from month_calculation import get_monthly_search_str
+import requirements
 from psql import connect_to_db
 from psql import insert_to_app
 from psql import insert_to_dependencies
 from psql import insert_to_package
 
-HEADERS = {"Authorization": "Bearer " + os.environ['GH_TOKEN']}
+HEADERS = {"Authorization": "Bearer " + os.environ['GITHUB_TOKEN']}
```
Comment (djsauble) [inline pipeline/scraper/github_query.py:16]: We're switching back to `GH_TOKEN`

File: pipeline/scraper/pypi_query.py [hunk 4]
```diff
@@ -56,12 +57,30 @@ def get_package_metadata(dependency):
...
 
-get_package_metadata('pkg:pypi/gunicorn@0.1')
\ No newline at end of file
+    entry['monthly_downloads_last_month'] = result['data']['last_month']
+    entry['monthly_downloads_a_year_ago'] = 0
+
+    try:
+        request_url = f'{PYPI_DEPENDENCY_META_URL}{dependency_name}/json'
+        json_result = requests.get(request_url).json()
+    except:
```
Comment (djsauble) [inline pipeline/scraper/pypi_query.py:67]: Would be nice to print the exception message for debugging purposes.

File: pipeline/scraper/github_query.py [hunk 5]
```diff
@@ -92,24 +98,27 @@ def run_query(today, language='JavaScript'):
 
     # fetch data and write to database
     last_node = None
-    monthly_search_str = get_monthly_search_str(today)
```
Comment (djsauble) [inline pipeline/scraper/github_query.py:95]: Do we need `get_monthly_search_str()` and related code anymore? Just remove that whole file?

File: pipeline/scraper/pypi_query.py (hunk 4 above)
Comment (vpopil) [reply pipeline/scraper/pypi_query.py:67]: Also, bare exception is not optimal. pylint or even pycodestyle will be screaming at this line

File: pipeline/model/generate.py [hunk 6]
```diff
@@ -46,7 +46,7 @@
 ARRAY = [Vectors.fromML(row.packages_encoded) for row in TRANSFORMED_DF.collect()]
 
 # Create a RowMatrix
-MATRIX = RowMatrix(SC.parallelize(ARRAY))
+MATRIX = RowMatrix(SC.parallelize(ARRAY, numSlices=50))
```
Comment (vpopil) [inline pipeline/model/generate.py:49]: 50 could be a CONST_PARAM

File: pipeline/scraper/github_query.py [hunk 7]
```diff
@@ -62,25 +65,28 @@ def write_db(database, result, language="JavaScript"):
...
                     dependency_str = 'pkg:pypi/' + package_name + "@" + package_version
                     package_id = insert_to_package(database, dependency_str)
+                    print(package_id)
                     insert_to_dependencies(database, str(application_id), str(package_id))
-                except:
-                    continue
-    database.commit()
 
+            except:
+                print(node['url'])
```
Comment (vpopil) [inline pipeline/scraper/github_query.py:88]: would it make sense to add a message that this has failed?

File: .github/workflows/test_pipeline.yml (hunk 1 above)
Comment (luke777gu) [reply .github/workflows/test_pipeline.yml:43]: Will do it after the merge

**Pull request merged**